### PR TITLE
Adding DataSet Class for Data I/O

### PR DESCRIPTION
**Pull request by stompsjo**
This implements a DataSet class with two methods important for downstream analysis:
- init_database: which pulls the necessary indexing information needed for data analysis from *one* `.h5` file.
- data_slice: which indexes a subset of data for analysis.

These methods utilize h5py for data i/o.

I would appreciate some design feedback on my class constructor, particularly whether the class contains sufficient attributes for its scope.

This PR also adds GitHub actions and code coverage support(?)

### CODE REVIEW

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: Test with pytest
+      run: |
+        pytest
+
+  coverage:
+    runs-on: ubuntu-latest
+    steps:
+      - checkout
+      - run: pip install -U pytest networkx matplotlib scipy coverage
```
Comment (gonuke) [inline .github/workflows/python-package.yml]: Are you using networks?

File: RadClass/DataSet.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    def __init__(self, labels = ['2x4x16LiveTimes', '2x4x16Times', '2x4x16Spectra']):
+        '''
+        Initializes DataSet class.
+        No vectors/data will be stored until methods are called.
+        '''
+
+        self.live = None
+        self.timestamps = None
+
+        self.live_label = labels[0]
```
Comment (gonuke) [inline RadClass/DataSet.py]: If the different labels have important semantic purposes, it's probably better to pass them as separate parameters, or a dictionary

File: RadClass/DataSet.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+        file = h5py.File(filename, 'r')
+        
+        # [:] returns a numpy array via h5py
+        self.live = file[datapath + self.live_label][:]
+        self.timestamps = file[datapath + self.timestamps_label][:]
+
+        file.close()
+
+        return self.live, self.timestamps
```
Comment (gonuke) [inline RadClass/DataSet.py]: Do you need to return these? The member variables have been set, so it's not clear where else you'll need the data

File: tests/test_load_data.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import numpy as np
+import os
+
+import RadClass.DataSet as ds
+from tests.create_file import create_file
+
+def test_init_database():
+    filename = 'testfile.h5'
+    datapath = '/uppergroup/lowergroup/'
+    labels = ['live', 'timestamps', 'spectra']
```
Comment (gonuke) [inline tests/test_load_data.py]: Dictionary for semantic clarity

File: tests/test_load_data.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    # checks if arrays were saved to file and read back correctly
+    np.testing.assert_almost_equal(test_live, live, decimal = 5)
+    np.testing.assert_almost_equal(test_timestamps, timestamps, decimal = 5)
+
+def test_data_slice():
+    # Creating sample dataset
+    filename = 'testfile.h5'
+    datapath = '/uppergroup/lowergroup/'
+    labels = ['live', 'timestamps', 'spectra']
```
Comment (gonuke) [inline tests/test_load_data.py]: Dictionary for semantic clarity

**Review by gonuke — CHANGES_REQUESTED** [review]
Plenty of little design aspects to discuss here

File: RadClass/DataSet.py (hunk 3 above)
Comment (stompsjo) [reply RadClass/DataSet.py]: Yeah...I was grappling with this when I was turning these two functions into class methods. I wasn't sure where this data should live (i.e. whether RadClass/AnalysisParameters should have a copy). Now it's clear that the data arrays/vectors should _stay_ in DataSet and just be accessed by other classes. ~~Maybe it is more appropriate to create getter methods.~~ (access attributes directly) Pushing now!

File: RadClass/DataSet.py (hunk 2 above)
Comment (stompsjo) [reply RadClass/DataSet.py]: Agreed, I think a dictionary would be appropriate. Although that means I will need a standardized set of keys for a labels dictionary. Pushing an update, let me know what you think...

**Comment by stompsjo** [comment]
Most recent push cleans python-package.yml, changes `label` list to a dictionary, and removes return statement from `init_database` method.

I should note that coverage and GitHub Actions aren't working just yet, I think because the .yml files must be in main first(?) Once this PR is merged, I can start an issue/PR for fixing CI. In the meantime, I am using `pytest` locally and both tests are passing.

**Review by gonuke — APPROVED** [review]
LGTM - thanks @stompsjo

**Pull request merged**